torchtext/datasets/imdb.py

```diff
@@ -23,16 +28,35 @@
 @_add_docstring_header(num_lines=NUM_LINES, num_classes=2)
 @_create_dataset_directory(dataset_name=DATASET_NAME)
 @_wrap_split_argument(('train', 'test'))
-def IMDB(root, split):
-    def generate_imdb_data(key, extracted_files):
-        for fname in extracted_files:
-            *_, split, label, file = Path(fname).parts
-
-            if key == split and (label in ['pos', 'neg']):
-                with io.open(fname, encoding="utf8") as f:
-                    yield label, f.read()
-    dataset_tar = download_from_url(URL, root=root,
-                                    hash_value=MD5, hash_type='md5')
-    extracted_files = extract_archive(dataset_tar)
-    iterator = generate_imdb_data(split, extracted_files)
-    return _RawTextIterableDataset(DATASET_NAME, NUM_LINES[split], iterator)
+def IMDB(root: str, split: Union[Tuple[str], str]):
+    if not is_module_available("torchdata"):
+        raise ModuleNotFoundError("Package `torchdata` not found. Please install following instructions at `https://github.com/pytorch/data`")
+
+    url_dp = IterableWrapper([URL])
+
+    cache_compressed_dp = url_dp.on_disk_cache(
+        filepath_fn=lambda x: os.path.join(root, _PATH),
+        hash_dict={os.path.join(root, _PATH): MD5}, hash_type="md5"
+    )
+    cache_compressed_dp = HttpReader(cache_compressed_dp).end_caching(mode="wb", same_filepath_fn=True)
+
+    labels = {"neg", "pos"}
+    cache_decompressed_dp = cache_compressed_dp.on_disk_cache(filepath_fn=lambda x: [os.path.join(root, "aclImdb", split, label) for label in labels])
+    cache_decompressed_dp = FileOpener(cache_decompressed_dp, mode="b")
+    cache_decompressed_dp = cache_decompressed_dp.read_from_tar()
+
+    def filter_imdb_data(key, fname):
+        # eg. fname = "aclImdb/train/neg/12416_3.txt"
+        *_, split, label, file = Path(fname).parts
+        return key == split and label in labels
+
+    cache_decompressed_dp = cache_decompressed_dp.filter(lambda t: filter_imdb_data(split, t[0]))
+
+    cache_decompressed_dp = cache_decompressed_dp.map(lambda t: (Path(t[0]).parts[-2], t[1]))  # eg. "aclImdb/train/neg/12416_3.txt" -> "neg"
```

Comment:
nit: update comment to say `"aclImdb_v1"`

Reply:
For this one, `aclImdb` refers to the actual folder name inside the tar. I'll keep it as is if there's no objections

Reply:
Gotcha that makes sense